ubiquity/dell-bootstrap.py

```diff
@@ -760,6 +761,16 @@ def prepare(self, unfiltered=False):
             self.log("rec_type %s, stage %d, device %s" % (rec_type, self.stage, self.device))
             if (rec_type == 'factory' and self.stage == 2) or rec_type == 'hdd':
                 self.fixup_factory_devices(rec_part)
+            if rec_type == 'hdd':
+                # copy old mok key so that user don't need to enroll it again.
+                rootfs = mount[0:-1] + '3'
+                self.log("old rootfs from %s" % rootfs)
+                misc.execute_root('mount', '-o', 'ro', rootfs, '/mnt')
```

Comment:
I think some extra sanity checking will be needed here.  Maybe check the filesystem?  If the installer key is run on a system with Windows 10, this will mount one of the Windows NTFS partitions needlessly.

Reply:
Even worse - if that partition is corrupt it will fail to mount and I don't believe the installer will be able to continue.

Reply:
the operation to keep key is expected to be done only when the user executes dell-recovery on well installed Ubuntu system. So, it should not happen to windows NTFS partition.

But a corruption partition could be a problem, any suggestion? Or maybe to force return true to not blocking process?

Reply:
Oh, I forgot you had this clause behind `recovery_type='hdd'`.  So actually if someone tries to run a Ubuntu recovery key on an Ubuntu system rather than recover from HDD they will lose the key that means.

So I think what I'm meaning is this should be used more widely, but it should be safe guarded.
I would recommend if corrupted partition to return true and not block process indeed.